src/Util/PaginationHelper.php

```diff
@@ -0,0 +1,197 @@
+<?php
+namespace ApiPlatform\Core\Util;
+
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\ResourceMetadata;
+use Exception;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+
+class PaginationHelper
+{
+    private $requestStack;
+    private $resourceMetadataFactory;
+    private $resourceMetadata;
+
+    private $enabled = true;
+    private $clientEnabled = false;
+    private $clientItemsPerPage = false;
+    private $itemsPerPage = 30;
+    private $maximumItemsPerPage = null;
+
+    private $parameterNamePage = 'page';
+    private $parameterNameEnabled = 'pagination';
+    private $parameterNameItemsPerPage = 'itemsPerPage';
+
+    private $resourceClass;
+    private $operationName;
+
+
+    /**
+     * PaginationHelper constructor.
+     * @param RequestStack $requestStack
+     * @param ResourceMetadataFactoryInterface $resourceMetadataFactory
+     */
+    public function __construct(RequestStack $requestStack, ResourceMetadataFactoryInterface $resourceMetadataFactory)
+    {
+        $this->requestStack = $requestStack;
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+    }
+
+
+    /**
+     * @param bool $enabled
+     * @param bool $clientEnabled
+     * @param bool $clientItemsPerPage
+     * @param int $itemsPerPage
+     * @param string $parameterNamePage
+     * @param string $parameterNameEnabled
+     * @param string $parameterNameItemsPerPage
+     * @param int|null $maximumItemsPerPage
+     */
+    public function setConfig(bool $enabled = true, bool $clientEnabled = false, bool $clientItemsPerPage = false, int $itemsPerPage = 30, string $parameterNamePage = 'page', string $parameterNameEnabled = 'pagination', string $parameterNameItemsPerPage = 'itemsPerPage', int $maximumItemsPerPage = null)
+    {
+        $this->enabled = $enabled;
+        $this->clientEnabled = $clientEnabled;
+        $this->clientItemsPerPage = $clientItemsPerPage;
+        $this->itemsPerPage = $itemsPerPage;
+        $this->maximumItemsPerPage = $maximumItemsPerPage;
+
+        $this->parameterNamePage = $parameterNamePage;
+        $this->parameterNameEnabled = $parameterNameEnabled;
+        $this->parameterNameItemsPerPage = $parameterNameItemsPerPage;
+    }
+
+
+    /**
+     * @param $name
+     * @return null
+     */
+    public function getConfig($name)
+    {
+        if (!in_array($name, ['enabled', 'clientEnabled', 'clientItemsPerPage', 'itemsPerPage', 'maximumItemsPerPage', 'parameterNamePage', 'parameterNameEnabled', 'parameterNameItemsPerPage'])) {
+            return null;
+        }
+
+        return $this->$name;
+    }
+
+
+    /**
+     * @param string $resourceClass
+     * @param string|null $operationName
+     */
+    public function setResourceAndOperation(string $resourceClass, string $operationName = null)
+    {
+        $this->resourceClass = $resourceClass;
+        $this->operationName = $operationName;
+    }
```

Comment:
The other way would be to supply these two parameters in every other method.
Should I change it that way?